### PR TITLE
TMA-807: ContextManager, SplunkLogger and GdLogger

### PR DESCRIPTION
**Pull request by kurecka**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/966/ : SUCCESS in 8s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/217/ : FAILURE in 1m 21s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/967/ : SUCCESS in 10s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/218/ : FAILURE in 1m 14s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/969/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/220/ : FAILURE in 1m 11s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/970/ : SUCCESS in 7s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/221/ : FAILURE in 1m 24s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/971/ : SUCCESS in 7s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/222/ : FAILURE in 1m 26s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/972/ : SUCCESS in 8s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/223/ : FAILURE in 1m 26s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/973/ : SUCCESS in 9s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/224/ : FAILURE in 1m 26s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/974/ : SUCCESS in 7s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/225/ : FAILURE in 1m 16s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/975/ : SUCCESS in 7s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/226/ : FAILURE in 1m 10s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/976/ : SUCCESS in 8s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/227/ : FAILURE in 1m 18s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/977/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/228/ : FAILURE in 1m 19s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/978/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/229/ : FAILURE in 1m 17s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/979/ : SUCCESS in 7s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/230/ : FAILURE in 1m 08s

File: lib/gooddata/core/splunk_logger.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+module GoodData
+  # Logger that process given message to format readable by splunk
+  class SplunkLogger < Logger
+    def hash_to_string(hash)
+      hash.map { |pair| " #{pair[0]}=#{pair[1]}" }.join ""
+    end
+
+    # If the given message or progname is an instance of Hash, it's reformatted to splunk readable format.
+    # In case that the message or the progname contain new line character log won't be printed out.
```
Comment (kubamahnert) [inline lib/gooddata/core/splunk_logger.rb]: what?

File: spec/lcm/integration/splunk_spec.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+require 'securerandom'
+
+def user_in_domain(user_name)
+  domain = @rest_client.domain(LcmConnectionHelper.environment[:prod_organization])
+  domain.find_user_by_login(user_name)
+end
+
+describe GoodData::SplunkLogger do
+  before(:each) do
+    @suffix = ConfigurationHelper.suffix
```
Comment (kubamahnert) [inline spec/lcm/integration/splunk_spec.rb]: could some of the new test structures (project_fixtures class maybe) be used here?

File: spec/lcm/integration/splunk_spec.rb [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      GoodData::Bricks::Pipeline.users_brick_pipeline.call(params)
+      expect(GoodData.gd_logger.logging_on?(:splunk)).to be_falsey
+    end
+  end
+
+  context 'when using mode sync_domain_and_project' do
+    prefix = SecureRandom.urlsafe_base64(16)
+    let(:file_name) { "splunk_#{prefix}.log" }
+
+    it 'adds users to domain' do
```
Comment (kubamahnert) [inline spec/lcm/integration/splunk_spec.rb]: this does not really expect to add the users to the domain, but rather checks that a "INFO" log happened to the file.

I would take two steps:

1) rephrase the 'it' description so it is more fitting of what the test expects to happen
2) actually test something in the test! Maybe check for a specific log message that should be always there?

**Comment by kubamahnert** [comment]
closed in favor of #1384

**Pull request closed without merging**